crates/circuit/src/circuit_data.rs

```diff
@@ -899,16 +899,7 @@ impl CircuitData {
             }
             let py_op = func.call1((self.unpack_py_op(py, instr)?,))?;
             let result = py_op.extract::<OperationFromPython>()?;
-            let params = match result.params {
-                Some(Parameters::Blocks(circuits)) => Some(Box::new(Parameters::Blocks(
-                    circuits
-                        .into_iter()
-                        .map(|circuit| self.blocks.push(circuit))
-                        .collect(),
-                ))),
-                Some(Parameters::Params(params)) => Some(Box::new(Parameters::Params(params))),
-                None => None,
-            };
+            let params = self.extract_blocks_from_circuit_parameters(result.params.as_ref());
```

Comment:
This newly clones the `Parameters::Params` variant (and the `Blocks` variant by ref), is that something we'd want to try and avoid, or is that by design and I'm not seeing why? 🤔

Reply:
We don't have to put a lot of effort now, but maybe just something to keep in mind when we're trying to optimize this flow later on

Reply:
I suspect I'm going to modify this in #15432 now, but I'm not 100% sure.